scripts/test.frontend.reproducibility.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# A script that runs frontend build in a row and compare the hashes of the generated files.
+# Exit if a difference is detected.
+
+DIR=tmp
+
+if [ ! -d "$DIR" ]; then
+  mkdir "$DIR"
+fi
+
+for ((n = 0; n < 10; n++)); do
+  npm run build && find build -type f -exec sha256sum {} \; | awk '{print $2 " " $1}' | sort >"$DIR/sha256-batch${n}.txt"
```

Comment:
Note that building several times in a row does not guarantee reproducibility.  For example, at one time Java used an implementation of HashMap that used a random seed set on boot.  This was a defence against targeted denial of service attacks that relied on knowing the HashMap buckets.  So as  long as a server was not rebooted, the elements of a hash map would be in the same order.  Reboot and they would be completely different.

This is still valuable, please just be aware that we might still have problems! :construction_worker:

Reply:
I hope we do not have a random seed hardcoded in the frontend 😉

Noted